specification/confidentialledger/Microsoft.mCCF/acks.tsp

```diff
@@ -0,0 +1,49 @@
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+
+import "./common.tsp";
+import "./cose.tsp";
+
+@TypeSpec.Versioning.useDependency(Azure.Core.Versions.v1_0_Preview_2)
+namespace Acks;
```

Comment:
@tjprescott This pattern came from the discussion here: https://github.com/microsoft/typespec/issues/1953

Moving this to a sub namespace results in duplicate definitions. ie in this case, I want the `GET /gov/members/state-digests/...` operations emitted from the `route("/gov/members") ... extends Acks {}`, but if this becomes a subnamespace it result in duplicate `GET /state-digests/...` operations being emitted.

Requiring that everything must be in a subnamespace to be emitted/documented forbids writing composable definitions across multiple namespaces. If the original concern was about `Azure.Core` types being included everywhere, can't we add configuration options to control which namespaces are (or aren't) considered for documentation, alongside the service namespace?

Reply:
I would argue that the underlying issue is that the duplicate definitions you cite is a bug of the openAPI emitter and that "putting it outside the service namespace" is a workaround, not a solution. For example, `Acks` as a namespace does not make any sense--it is closely tied to your service.

Imagine a scenario where every service starts putting their models in `namespace Models`. That will easily become polluted. This spec is putting its models in the global namespace, which is arguably even worse than putting it in the `Models` namespace.

cc/ @johanste

Reply:
Thanks for caring

Reply:
I've found a fix that seems a reasonable compromise, which is to just declare the full implementation details at all of the source interfaces. So rather than
```
// foo.tsp
interface AbstractFoo { }

// main.tsp
@service(...)
@versioned(...)
namespace MyService
{
  @route(...)
  @otherDecoratorsForActualImpl(...)
  interface NicelyNamedFoo extends AbstractFoo { }
}
```

I can do
```
// foo.tsp
namespace MyService
{
  @route(...)
  @otherDecoratorsForActualImpl(...)
  interface NicelyNamedFoo { }
}

// main.tsp
@service(...)
@versioned(...)
namespace MyService;
```

This means some repetition (eg - where I have multiple interfaces on the same route prefix, I need to repeat that prefix on each interface rather than combining them in a single "implementation" interface), but at least gives consistent naming and means everything is namespaced and so not double-emitted.

I'm surprised that content in the global namespace is viewed as pollution rather than implementation. I was writing my spec as though its a module in JS or Python, where there's some explicit _export_ of names that anything else might see, and outside of those it's fine to work (to build abstractions and components and details) in the global or common namespaces. The `@service` decorator seems to function as an export statement, declaring what names and types should be exported. If the intent is instead that there is no room for private details/machinery in TypeSpec (or at least Azure TypeSpec), and everything is imported and visible, then I agree it should all be namespaced to avoid polluting top-level names. It sounds like that recommendation deserves a linter warning - every definition _must_ be under a service subnamespace.

Reply:
I've moved all of the TypeSpec definitions under the `Microsoft.ManagedCcf` namespace, and the generated ApiView now describes the content of this PR. Thanks for your help @tjprescott.